L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc

```diff
@@ -0,0 +1,204 @@
+#include "L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "TCanvas.h"
+#include "TFile.h"
+#include "TH1.h"
+#include "TText.h"
+#include "TPaveLabel.h"
+#include "TLegend.h"
+#include "TPostScript.h"
+#include "TStyle.h"
+#include "TROOT.h"
+#include "TFile.h"
+#include "TStyle.h"
+#include "TLatex.h"
+
+#include <iostream>
+CSCTriggerPrimitivesAnalyzer::CSCTriggerPrimitivesAnalyzer(const edm::ParameterSet &conf)
+    : rootFileName_(conf.getParameter<std::string>("rootFileName")),
+      runNumber_(conf.getParameter<unsigned>("runNumber")),
+      monitorDir_(conf.getParameter<std::string>("monitorDir")),
+      chambers_(conf.getParameter<std::vector<std::string>>("chambers")),
+      alctVars_(conf.getParameter<std::vector<std::string>>("alctVars")),
+      clctVars_(conf.getParameter<std::vector<std::string>>("clctVars")),
+      lctVars_(conf.getParameter<std::vector<std::string>>("lctVars")),
+      dataVsEmulatorPlots_(conf.getParameter<bool>("dataVsEmulatorPlots")),
+      mcEfficiencyPlots_(conf.getParameter<bool>("mcEfficiencyPlots")),
+      mcResolutionPlots_(conf.getParameter<bool>("mcResolutionPlots")) {}
+
+void CSCTriggerPrimitivesAnalyzer::analyze(const edm::Event &ev, const edm::EventSetup &setup) {
+  // efficiency and resolution analysis is done here
+}
+
+void CSCTriggerPrimitivesAnalyzer::endJob() {
+  if (dataVsEmulatorPlots_)
+    makeDataVsEmulatorPlots();
+}
+
+void CSCTriggerPrimitivesAnalyzer::makeDataVsEmulatorPlots() {
+  // data vs emulator plots are created here
+  edm::Service<TFileService> fs;
+
+  // split monitorDir_ into two substrings
+  std::string delimiter = "/";
+  int pos = monitorDir_.find(delimiter);
+  std::string superDir = monitorDir_.substr(0, pos);
+  monitorDir_.erase(0, pos + delimiter.length());
+  std::string subDir = monitorDir_;
+  std::string path = "DQMData/Run " + std::to_string(runNumber_) + "/" + superDir + "/Run summary/" + subDir + "/";
+
+  std::cout << superDir << " " << subDir << std::endl;
+
+  TFile *theFile = new TFile(rootFileName_.c_str());
+  if (theFile) {
+    std::cout << "Loading DQM ROOT file: " << rootFileName_ << std::endl;
```

Comment:
`cout` is not thread safe, you should use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger.
You can use `cout` if it is simply a test code

Reply:
Removed or replaced by LogError